### PR TITLE
Add support for UPDATE statement

### PR DESCRIPTION
**Pull request by zeninpalm**

### CODE REVIEW

File: sqlglot/parser.py [hunk 1]
```diff
@@ -327,6 +330,22 @@ def _parse_insert(self):
             overwrite=overwrite,
         )
 
+    def _parse_update(self):
+        self._match(TokenType.UPDATE)
```
Comment (tobymao) [inline sqlglot/parser.py:334]: you don't need this extra match here, you already do it on line 243

File: tests/test_expressions.py [hunk 2]
```diff
@@ -43,6 +43,9 @@ def test_find_all(self):
     def test_sql(self):
         assert parse('x + y * 2')[0].sql() == 'x + y * 2'
         assert parse('select "x"')[0].sql(dialect='hive', pretty=True) == 'SELECT\n  `x`'
+        assert parse('update tbl_name set foo = 123')[0].sql() == 'UPDATE tbl_name SET foo = 123'
```
Comment (tobymao) [inline tests/test_expressions.py:46]: can you delete these tests and instead add them to identity?

also, add a test case for set x = 1, y =2

File: sqlglot/expressions.py [hunk 3]
```diff
@@ -202,6 +202,11 @@ class Unnest(Expression):
     arg_types = {'expressions': True, 'ordinality': False, 'table': False, 'columns': False}
 
 
+class Update(Expression):
+    token_type = TokenType.UPDATE
+    arg_types = {'this': True, 'set': True, 'where': False}
```
Comment (tobymao) [inline sqlglot/expressions.py:207]: maybe change set to be expressions since we're going to need to do CSV

File: sqlglot/parser.py [hunk 4]
```diff
@@ -327,6 +330,22 @@ def _parse_insert(self):
...
+        this = self._parse_table(None)
+
+        self._match(TokenType.SET)
+        set_expression = self._parse_expression()
```
Comment (tobymao) [inline sqlglot/parser.py:338]: one statement

return exp.Update(
  this=self._parse_table(None),
  expressions=self._parse_csv(self._parse_equality),
  where=self._parse_where(),
)

File: sqlglot/generator.py [hunk 5]
```diff
@@ -209,6 +209,16 @@ def table_sql(self, expression):
     def tuple_sql(self, expression):
         return f"({self.expressions(expression, flat=True)})"
 
+    def update_sql(self, expression):
+        this = self.sql(expression, 'this')
+        set_exp = expression.args.get('set')
+        set_sql = self.sql(set_exp)
```
Comment (tobymao) [inline sqlglot/generator.py:215]: probably cleaned up a bit

set_sql = self.expressions(expression)

File: sqlglot/parser.py (hunk 1 above)
Comment (zeninpalm) [reply sqlglot/parser.py]: Thanks, I didn't notice that.

File: sqlglot/expressions.py (hunk 3 above)
Comment (zeninpalm) [reply sqlglot/expressions.py]: Sure, I've changed it to expressions

File: sqlglot/generator.py (hunk 5 above)
Comment (zeninpalm) [reply sqlglot/generator.py]: Sure, I've cleaned it up.

File: tests/test_expressions.py (hunk 2 above)
Comment (zeninpalm) [reply tests/test_expressions.py]: Thanks, I've moved the test cases to identity

File: sqlglot/parser.py (hunk 4 above)
Comment (zeninpalm) [reply sqlglot/parser.py]: Great idea 😄

**Review by tobymao — APPROVED** [review]

**Pull request merged**